content/docs/iac/concepts/debugging/_index.md

```diff
@@ -20,288 +20,61 @@ Because Pulumi uses general purpose programming languages to provision cloud res
-
-```bash
-$ pulumi up
-Previewing update (dev)
-
- Type                 Name  Plan   Info
- pulumi:pulumi:Stack  dev          0.00s - Note: Debugging will proceed.
-```
-
-Open your Pulumi program in your IDE and set a breakpoint at the line where you want the debugger to stop.
+Pulumi provides an extension that allows you to launch and debug Pulumi programs with a few simple steps.
 
-Navigate to your IDE and attach the debugger to the process with the `python` runtime. For example, in [VS Code](https://code.visualstudio.com/docs/editor/debugging), you can define the following `.vscode/launch.json` file:
+### Install the extension
 
-```json
-{
-    "version": "0.2.0",
-    "configurations": [
-        {
-            "name": "Python: Local Attach",
-            "type": "debugpy",
-            "request": "attach",
-            "connect": {
-                "host": "localhost",
-                "port": 5678
-            },
-            "pathMappings": [
-                {
-                    "localRoot": "${workspaceFolder}",
-                    "remoteRoot": "."
-                }
-            ],
-            "justMyCode": true
-        }
-    ]
-}
-```
+Install the [Pulumi extension](https://marketplace.visualstudio.com/items?itemName=pulumi.pulumi-vscode-tools) using Visual Studio Marketplace.
 
-Navigate to the `Run and Debug` menu on the left pane, and then click the Start Debugging (F5) button with the `Python: Local Attach` configuration.
+### Open a project
 
-Now you can step through the program and inspect variables. Once completed, resume the execution and let the Pulumi CLI complete.
+Open a new or existing Pulumi project as a VS Code workspace. The extension supports both [single-folder workspaces](https://code.visualstudio.com/docs/editor/workspaces#_singlefolder-workspaces)
+and [multi-root workspaces](https://code.visualstudio.com/docs/editor/workspaces#_multiroot-workspaces).
 
-## Debugging .NET programs
+### Start Debugging
 
-Prepend your Pulumi program with the code to wait for a debugger to attach:
+Pulumi programs are run (with or without debugging) using a [launch configuration](https://code.visualstudio.com/docs/editor/debugging#_launch-configurations). Select the Run and Debug icon in the Activity Bar on the side of VS Code, then use an automatic debug configuration or create a launch configuration file for your project.
 
-```cs
-using System.Diagnostics;
+The extension automatically generates a debug configuration to run `pulumi up` or `pulumi preview` for the current Pulumi stack. To use an automatic debug configuration, do the following:
 
-return await Pulumi.Deployment.RunAsync(async () => // make sure you have the async keyword here
-{
-    while (!Debugger.IsAttached)
-    {
-        // keep waiting until the debugger is attached
-        await System.Threading.Tasks.Task.Delay(1000);
-    }
+1. Select the __Run and Debug__ icon.
+2. Choose __Show all automatic debug configurations__.
```

Comment:
We may want to mention that a debuggable file like `index.ts` should also be open. I happened to have `Pulumi.dev.yaml` open when I clicked Run and Debug, and I got an error explaining that I didn't have a YAML debugging extension.